controller/k8s/pods.go

```diff
@@ -46,7 +46,7 @@ func NewPodIndex(clientset *kubernetes.Clientset, index cache.IndexFunc) (*PodIn
 
 func (p *PodIndex) Run() error {
 	go p.reflector.ListAndWatch(p.stopCh)
-	return newWatcher(p.reflector, podResource).run()
+	return newWatcher(p.reflector, podResource, p.reflector.ListAndWatch, p.stopCh).run()
```

Comment:
Since you're passing `p.reflector.ListAndWatch` into the `newWatcher` function now, you don't need to call it here on line 48. As written, the `ListAndWatch` function will be called twice: once line 48, and again on line 47 in watcher.go. I'd just remove the call on line 48 in this file:

```
func (p *PodIndex) Run() error {
	return newWatcher(p.reflector, podResource, p.reflector.ListAndWatch, p.stopCh).run()
}
```

Reply:
Thanks for that, that was an oversight on my part.